### PR TITLE
fix: reset widget to default  value after setter method 

### LINKED ISSUE #27119: [Bug]: `resetWidget()` Not Working for Input Widget Where `setValue()` Was Used
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When running the `resetWidget()` method for an Input widget for which we previously ran the `setValue()` method, the widget does not reset to its default value. Even after manually typing something else in the Input widget, if then running `resetWidget()`, it gets set to the value that was provided with the `setValue()` method, as if it has overwritten the default value.
However, on page refresh, the **Default Value** is shown again in the input widget as expected. It seems that this is the only way to reset the Input widget to its default value after the `setValue()` method was used.

### Steps To Reproduce

1. Add an **Input widget** called `Input1` to the canvas.
2. Enter a **Default Value** for it (e.g., `John`).
3. Add a **Button widget** to the canvas and set its **onClick** method to `{{Input1.setValue('Hello!')}}`.
4. Add another **Button widget** to the canvas and set its **OnClick** method to `{{resetWidget("Input1")}}`.
5. Click the **first button** that sets the value.
6. Click the **second button** that is supposed to reset it.
7. Type something else in the **Input widget**.
8. Click again the **button** that is supposed to reset it.
9. Refresh the page.

### Public Sample App

_No response_

### Environment

Production

### Issue video log

https://www.loom.com/share/a514cfec83cf4e2f854e1a63312c40f1

### Version

Appsmith v1.9.36

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description
This PR ensures that widgets are reset to their default value after the `setValue` setter method is used to set its value

#### PR fixes following issue(s)
Fixes #27119 

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Implemented a new reset functionality for widgets, allowing users to revert to default values after changes.
  - Enhanced widget meta updates with a new reset action.

- **Bug Fixes**
  - Added test cases to ensure widget reset functionality works as expected, even after asynchronous operations.

- **Refactor**
  - Refactored evaluation logic to improve handling of widget meta updates and resets.
  - Improved action execution logic for resetting widget properties.

- **Tests**
  - Expanded end-to-end regression tests to cover new reset widget functionality.

- **Documentation**
  - Updated internal documentation to reflect new action types and evaluation processes related to widget meta updates.

### CODE REVIEW

**Review by arunvjn — APPROVED** [review]

File: app/client/src/workers/Evaluation/fns/resetWidget.ts [hunk 1]
```diff
@@ -1,22 +1,284 @@
...
+        const expressionToEvaluate: string = unEvalEntity[defaultPropertyPath];
+
+        let finalValue: unknown;
+        if (
+          expressionToEvaluate &&
+          typeof expressionToEvaluate === "string" &&
+          isDynamicValue(expressionToEvaluate)
+        ) {
+          const { jsSnippets } = getDynamicBindings(expressionToEvaluate);
+          const { result } = evaluateSync(
```
Comment (rishabhrathod01) [inline app/client/src/workers/Evaluation/fns/resetWidget.ts:107]: evaluateSync was added here to re-evaluate the default value of the widget and then use that to reset. This was done as we want to widget to go back to default value once reset. All of this was due to setter methods were changing the default value. 

`DefaultText : {{appsmith.user.email}} `

```
Input1.setText("some random value") // this line changes the default text of input to "some random value
```

When we reset Input1 we want to reset back to `{{appsmith.user.email}}` hence we re-evaluate the field and then set it. In this process of re-evaluation we use sync eval operation.

**Pull request merged**